internal/controller/atlasproject/custom_roles.go

```diff
@@ -50,16 +50,20 @@ func getLastAppliedCustomRoles(atlasProject *akov2.AtlasProject) ([]akov2.Custom
 	return lastAppliedSpec.CustomRoles, nil
 }
 
-func findRolesToDelete(prevSpec, atlasRoles []customroles.CustomRole) map[string]customroles.CustomRole {
+func findRolesToDelete(prevSpec, akoRoles, atlasRoles []customroles.CustomRole) map[string]customroles.CustomRole {
 	result := map[string]customroles.CustomRole{}
-	for atlasRoleIdx := range atlasRoles {
-		for specRoleIdx := range prevSpec {
-			if atlasRoles[atlasRoleIdx].Name == prevSpec[specRoleIdx].Name {
-				result[prevSpec[specRoleIdx].Name] = prevSpec[specRoleIdx]
-				continue
-			}
+	lastAppliedRolesMap := mapCustomRolesByName(prevSpec)
+	akoRolesMap := mapCustomRolesByName(akoRoles)
+	atlasRolesMap := mapCustomRolesByName(atlasRoles)
+
+	for atlasIx, atlasRole := range atlasRolesMap {
+		_, inAKO := akoRolesMap[atlasIx]
+		_, inLastApplied := lastAppliedRolesMap[atlasIx]
+		if !inAKO && inLastApplied {
+			result[atlasIx] = atlasRole
```

Comment:
Why `Ix` instead of just `Name`?

```suggestion
	for atlasName, atlasRole := range atlasRolesMap {
		_, inAKO := akoRolesMap[atlasName]
		_, inLastApplied := lastAppliedRolesMap[atlasName]
		if !inAKO && inLastApplied {
			result[atlasName] = atlasRole
```

Reply:
it's the index of the map, but name sounds good too